src/test/ui/consts/const-err-late.rs

```diff
@@ -17,6 +17,4 @@ impl<T> S<T> {
 
 fn main() {
     black_box((S::<i32>::FOO, S::<u32>::FOO));
-    //~^ ERROR erroneous constant
-    //~| ERROR erroneous constant
```

Comment:
This is basically the reason the sometimes-duplicate error was added. Without it, it becomes unclear what caused the error

Reply:
There is still an error pointing at `FOO`. But the problem is the missing instantiation site?

This is a general problem with our const-eval errors -- they don't show where the const was used. Not sure what the best way is to add that information. After all the query is only evaluated once, without knowing the use sites. A separate lint is kind of a hack. Making this part of const_prop_lint is also a hack, this is an entirely separate concern. (Seems fine to keep it there I guess but there should be a comment explaining all this.)